File: gameoflife.php
<?php
// Copyright 2009 Raphael Michel <http://www.raphaelmichel.de>
// It's public domain. But please do not remove this note
// Usage:
// php gameoflife.php [timeout [width [height]]]
// Examples:
//      php gameoflife.php
//      php gameoflife.php 2
//      php gameoflife.php 5 60
//      php gameoflife.php 1 70 26
// Test Run:
//      php gameoflife.php glider

class gameoflife {
    // Attention: field[y][x]
    var $field = array();
    
    function initfield($width, $height, $rand = true){
        for($y = 0; $y < $height; $y++){
            for($x = 0; $x < $width; $x++){
                if($rand)
                    $this->field[$y][$x] = (bool) rand(0,1);
                else
                    $this->field[$y][$x] = false;
            }
        }
    }

    function print_field($field){
        for($y = 0; $y < count($field); $y++){
            for($x = 0; $x < count($field[$y]); $x++){
                if($field[$y][$x])
                    echo "X";
                else echo " ";
            }
            echo "\n";
        }   
    }

    function next_generation($field){
        $newfield = array();
        $this->any = false;
        for($y = 0; $y < count($field); $y++){
            for($x = 0; $x < count($field[$y]); $x++){
                $cell = $field[$y][$x];
                $neighbours = 0;
                if($field[($y-1)][($x-1)]) $neighbours++;
                if($field[($y-1)][($x)]) $neighbours++;
                if($field[($y)][($x-1)]) $neighbours++;
                if($field[($y+1)][($x+1)]) $neighbours++;
                if($field[($y+1)][($x)]) $neighbours++;
                if($field[($y)][($x+1)]) $neighbours++;
                if($field[($y-1)][($x+1)]) $neighbours++;
                if($field[($y+1)][($x-1)]) $neighbours++;
                
                if($cell && ($neighbours <= 1 || $neighbours >= 4)) $cell = false;
                elseif(!$cell && $neighbours == 3) $cell = true;
                else $cell = $cell;
                if(!$this->any && $cell) $this->any = true;
                $newfield[$y][$x] = $cell; 
            }
        } 
        return $newfield;
    }

    function startgame($width = 50, $height = 20, $timeout = 2, $init = true){
        if($init) $this->initfield($width, $height);
        $i = 1;
        while(1) {
            $i++;
            system(strtoupper(substr(PHP_OS, 0, 3)) === 'WIN' ? "cls" : "clear");
            $this->field = $this->next_generation($this->field);
            $this->print_field($this->field);
            echo "Generation $i | Conway's Game of Life | PHP-Random-Implementation";
            if(!$this->any) die("\nNothing left.");
            sleep($timeout);
        }
    }
}
$gol = new gameoflife;
if($argv[1] == 'glider'){ // to test whether it's a real game of life
    $gol->initfield(50, 20, false);
    $gol->field[5][8] = true;
    $gol->field[5][9] = true;
    $gol->field[5][10] = true;
    $gol->field[4][10] = true;
    $gol->field[3][9] = true;
    $gol->startgame((($argv[3]) ? $argv[3] : 50), (($argv[4]) ? $argv[4] : 20), (($argv[2]) ? $argv[2] : 2), false);
}else{
    $gol->startgame((($argv[2]) ? $argv[2] : 50), (($argv[3]) ? $argv[3] : 20), (($argv[1]) ? $argv[1] : 2));
}

File: gameoflife.py
#!/usr/bin/env python
# Copyright 2009 Raphael Michel <http://www.raphaelmichel.de>
# It's public domain. But please do not remove this note
# Usage:
# ./gameoflife.py [timeout [width [height]]]
# Examples:
#      ./gameoflife.py
#      ./gameoflife.py 2
#      ./gameoflife.py 5 60
#      ./gameoflife.py 1 70 26

import sys
import random
import os
import time

class gol:
    def initfield(self, width, height):
        field = [False] * height
        for i in range(0, height):
            field[i] = [False] * width
            
        for y in range(0, height):
            for x in range(0, width):
                if random.randint(0,1):
                    field[y][x] = True
                else:
                    field[y][x] = False
        return field
    
    def print_field(self, field, width, height):
        if os.name == 'nt':
            os.system('cls')
        else:
            os.system('clear')
        for y in range(0, height):
            for x in range(0, width):
                if field[y][x]:
                    sys.stdout.write(' ')
                else:
                    sys.stdout.write('X')
            print ""
    
    def nextgen(self, field, width, height):
        newfield = field
        for y in range(0, height):
            for x in range(0, width):
                neighbours = 0
                cell = field[y][x]
                try:
                    if field[y-1][x-1]:
                        neighbours += 1
                except:
                    a = 1
                try:
                    if field[y][x-1]:
                        neighbours += 1
                except:
                    a = 1
                try:
                    if field[y-1][x]:
                        neighbours += 1
                except:
                    a = 1
                try:
                    if field[y+1][x+1]:
                        neighbours += 1
                except:
                    a = 1
                try:
                    if field[y+1][x]:
                        neighbours += 1
                except:
                    a = 1
                try:
                    if field[y][x+1]:
                        neighbours += 1
                except:
                    a = 1
                try:
                    if field[y-1][x+1]:
                        neighbours += 1
                except:
                    a = 1
                try:
                    if field[y+1][x-1]:
                        neighbours += 1
                except:
                    a = 1
                
                if (cell and (neighbours <= 1 or neighbours >= 4)): cell = False
                elif (not cell and neighbours == 3): cell = True
                else: cell = cell
                
                newfield[y][x] = cell
        return newfield
    def start(self, width, height, timeout):
        i = 0
        field = self.initfield(width, height)
        while True:
            field = self.nextgen(field, width, height)
            self.print_field(field, width, height)
            print("Generation "+str(i)+" | Conway's Game of Life | Python-Random-Implementation")
            i = i+1
            time.sleep(timeout)

g = gol()
if len(sys.argv) == 1:
    width = 50
    height = 20
    timeout = 2
elif len(sys.argv) == 1:
    width = 50
    height = 20
    timeout = int(sys.argv[1])
elif len(sys.argv) == 2:
    width = int(sys.argv[2])
    height = 20
    timeout = int(sys.argv[1])
elif len(sys.argv) == 3:
    width = int(sys.argv[2])
    height = int(sys.argv[3])
    timeout = int(sys.argv[1])

g.start(width, height, timeout)
